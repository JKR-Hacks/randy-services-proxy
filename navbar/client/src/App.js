/* eslint-disable prefer-destructuring */
/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import axios from 'axios';

import Navigation from './components/Navigation';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      teams: [],
      view: 'main',
    };
    this.renderView = this.renderView.bind(this);
  }

  componentDidMount() {
    axios.get('http://localhost:3007/espn/teamstandings', {
      method: 'GET',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
    })
      .then((res) => {
        const data = res.data;
        this.setState({
          teams: data,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  // eslint-disable-next-line consistent-return
  renderView() {
    const { view, teams } = this.state;
    if (view === 'main') {
      return (
        <div>
          <Navigation teams={teams} />
        </div>
      );
    }
  }

  render() {
    return (
      this.renderView()
    );
  }
}
