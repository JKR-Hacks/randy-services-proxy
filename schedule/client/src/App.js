/* eslint-disable consistent-return */
import React, { Component } from 'react';
import axios from 'axios';

import Schedule from './components/Schedule';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      schedule: [],

      view: 'schedule',
    };
  }

  componentDidMount() {
    // fetch schedule data
    axios.get('http://localhost:3002/espn/schedules', {
      method: 'GET',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
    })
      .then((res) => {
        // eslint-disable-next-line prefer-destructuring
        const data = res.data;
        this.setState({
          schedule: data,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  renderView() {
    const { schedule, view } = this.state;
    if (view === 'schedule') {
      return (
        <div id="fullschedule">
          <Schedule
            ramsSchedule={schedule}
          />
        </div>
      );
    }
    // if (view === 'feed') {
    //   return (
    //     <div id="feed">
    //       <Feed
    //         ramsFeed={ramsFeed}
    //       />
    //     </div>
    //   );
    // }
    // if (view === 'sidebarSchedule') {
    //   return (
    //     <div id="sidebarSchedule">
    //       <SidebarSchedule
    //         ramsSchedule={schedule}
    //       />
    //     </div>
    //   );
    // }
  }

  render() {
    return (
      <div>
        {this.renderView()}
      </div>
    );
  }
}

export default App;
