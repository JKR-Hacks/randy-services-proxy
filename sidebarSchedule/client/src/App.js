import React, { Component } from 'react';
import axios from 'axios';

import SidebarSchedule from './components/SidebarSchedule';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      schedule: [],
      view: 'main',
    };
  }

  componentDidMount() {
    // fetch schedule data
    axios.get('http://localhost:3006/espn/schedules', {
      method: 'GET',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
    })
      .then((res) => {
        // eslint-disable-next-line prefer-destructuring
        const data = res.data;
        this.setState({
          schedule: data,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  renderView() {
    const { schedule, view } = this.state;
    if (view === 'main') {
      return (
        <div id="SidebarSchedule">
          <SidebarSchedule
            ramsSchedule={schedule}
          />
        </div>
      );
    }
  }

  render() {
    return (
      <div>
        {this.renderView()}
      </div>
    );
  }
}

export default App;
