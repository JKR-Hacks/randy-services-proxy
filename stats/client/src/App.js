/* eslint-disable import/extensions */
import React, { Component } from 'react';
import $ from 'jquery';
import MainComponent from './components/MainComponent.jsx';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sample: [],
    };
  }

  componentDidMount() {
    $.ajax({
      method: 'GET',
      url: 'http://localhost:3004/stats',
      mode: 'no-cors',
      headers: {
        "Access-Control-Allow-Origin": '*',
        'Content-Type': 'application/json',
      },
      error: (err) => {
        console.log(err, 'err');
      },
      success: (data) => {
        this.setState({
          sample: data,
        });
      },
    });
  }


  render() {
    const { sample } = this.state;
    return (
      <div>
        <MainComponent sample={sample} />
      </div>
    );
  }
}

export default App;
