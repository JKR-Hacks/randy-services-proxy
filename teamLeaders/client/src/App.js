/* eslint-disable react/jsx-no-bind */
/* eslint-disable import/extensions */
import React, { Component } from 'react';
import $ from 'jquery';
import Teamleaders from './components/Teamleaders.jsx';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sample: [],
      photos: [],
      side: 'Offense',
    };
  }

  componentDidMount() {
    $.ajax({
      method: 'GET',
      url: 'http://localhost:3003/stats',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
      error: (err) => {
        console.log(err, 'err');
      },
      success: (data) => {
        this.setState({
          sample: data,
        });
      },
    });
    $.ajax({
      method: 'GET',
      url: 'http://localhost:3003/photos',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
      dataType: 'json',
      error: (err) => {
        console.log(err, 'err');
      },
      success: (data) => {
        this.setState({
          photos: data,
        });
      },
    });
  }

  changeSide(e) {
    this.setState({
      side: e,
    });
  }

  render() {
    const { sample, photos, side } = this.state;
    return (
      <div>
        <Teamleaders
          sample={sample}
          photos={photos}
          changeSide={this.changeSide.bind(this)}
          side={side}
        />
      </div>
    );
  }
}

export default App;
