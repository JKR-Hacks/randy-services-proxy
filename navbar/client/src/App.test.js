/* eslint-disable no-unused-vars */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import axios from 'axios';

import App from './App';
import FullStandings from './components/FullStandings';

jest.mock('./components/FullStandings', () => 'fullstandings');

describe('App component', () => {
  it('Matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render App component correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should update a state in a callback', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ scores: 'new' }, () => {
      expect(wrapper.state()).toEqual({ scores: 'new', teams: [], view: 'main' });
    });
  });

  it('Should call componentDidMount once', () => {
    const spy = jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('Should update state view when handleClick() is called', () => {
    const wrapper = shallow(<App />);
    wrapper.instance().handleClick();
    expect(wrapper.state()).toEqual({ teams: [], view: 'fullstandings' });
  });
});

describe('FullStandings component', () => {
  it('Should render FullStandings component', () => {
    const tree = mount(<FullStandings />).toJSON;
    expect(tree).toMatchSnapshot();
  });
});

describe('Make successful API call to database', () => {
  it('Should fetch via axios /espn/teamstandings', () => {
    const spy = jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);
    wrapper.instance().componentDidMount(axios.get('/espn/teamstandings'));
    expect(spy).toHaveBeenCalled();
  });
});
