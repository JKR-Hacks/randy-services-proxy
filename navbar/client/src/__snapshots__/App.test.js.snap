// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches the snapshot 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt="logo"
          src="https://a.espncdn.com/i/teamlogos/nfl/500/lar.png"
          width="50px"
        />
      </a>
      <h3>
        LOS ANGELES RAMS
      </h3>
      <button
        className="btn btn-outline-primary btn-sm"
        type="button"
      >
        Follow
      </button>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarSupportedContent"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="nav-item active"
          >
            <a
              className="nav-link"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
            >
              Stats
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
            >
              Schedule
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
            >
              Standings
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div
    id="mainstandings"
  >
    <div
      className="container-fluid"
    >
      <h5>
        2018 NFC West Standings
      </h5>
      <table
        className="table table-striped table-hover table-sm rounded"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              Team
            </th>
            <th
              scope="col"
            >
              W
            </th>
            <th
              scope="col"
            >
              L
            </th>
            <th
              scope="col"
            >
              T
            </th>
            <th
              scope="col"
            >
              PCT
            </th>
            <th
              scope="col"
            >
              PF
            </th>
            <th
              scope="col"
            >
              PA
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
      <button
        className="btn btn-link"
        onClick={[Function]}
        type="button"
      >
        Full Standings
      </button>
    </div>
  </div>
</div>
`;

exports[`App component Should render App component correctly 1`] = `
<div>
  <Navigation
    teams={Array []}
  />
  <div
    id="mainstandings"
  >
    <Standings
      handleClick={[Function]}
      teams={Array []}
    />
  </div>
</div>
`;

exports[`FullStandings component Should render FullStandings component 1`] = `undefined`;
